pinot-core/src/main/java/com/linkedin/pinot/core/util/DataTableCustomSerDe.java

```diff
@@ -46,6 +48,26 @@
  */
 public class DataTableCustomSerDe extends DataTableJavaSerDe {
   private static final Logger LOGGER = LoggerFactory.getLogger(DataTableCustomSerDe.class);
+  private BrokerMetrics _brokerMetrics;
+
+  /**
+   * Default constructor for the class.
+   */
+  public DataTableCustomSerDe() {
+    _brokerMetrics = null;
+  }
+
+  /**
+   * Constructor for the class, registers the broker metrics to emit a metric each time
+   * we hit a Java serialization/de-serialization, as the goal is to catch all of them
+   * and replace with custom ser/de.
+   *
+   * @param brokerMetrics Broker metrics
+   */
+  public DataTableCustomSerDe(BrokerMetrics brokerMetrics) {
```

Comment:
I was suggesting that adding broker metrics to class in a core/utils seems incorrect. But can this class / method not return if the object was using java ser/de? That way you can increment the metric where the instance of this class is used. If it is too cumbersome and if this is temporary I am fine with it.